go.mod

```diff
@@ -142,4 +138,9 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace github.com/imdario/mergo => github.com/imdario/mergo v0.3.16 // Replaced so that 'go get -u' works. Remove/bump when upgrading.
+replace (
+	// Replaced for compatibility with k8s.io/component-base@v0.29.5
+	// Above v0.29.5, we'd need to upgrade Go to v1.22, which isn't compatible
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.18.0
+	github.com/prometheus/common => github.com/prometheus/common v0.47.0
+)
```

Comment:
Trades out replace statements, but I'm assuming the `kubernetes-dependency-watch` version bump is desireable, so hopefully the next round we can get rid of these entirely.